### PR TITLE
File Restructuring

### PR DESCRIPTION
**Pull request by b-scholz**
Some of the AST files had still Ast as a preamble. This PR removes the preamble to make the AST files in the src/ast directory consistency. 

We still need to break up some AST files into multiple ones and give them sensible names (cf. analysis and transforms).

### CODE REVIEW

**Comment by mmcgr** [comment]
Looking good, but src/tests/* needs updated include paths.

**Comment by b-scholz** [comment]
Thanks - I will also do the RAM.

File: src/Makefile.am [hunk 1]
```diff
@@ -235,23 +235,23 @@ souffle_sources = \
...
-        ram/RamUtils.h                                     \
-        ram/RamVisitor.h                                   \
-        ram/analysis/RamAnalysis.h                         \
-        ram/analysis/RamComplexityAnalysis.cpp             \
-        ram/analysis/RamComplexityAnalysis.h               \
-        ram/analysis/RamIndexAnalysis.cpp                  \
-        ram/analysis/RamIndexAnalysis.h                    \
-        ram/analysis/RamLevelAnalysis.cpp                  \
-        ram/analysis/RamLevelAnalysis.h                    \
+        ram/Condition.h                                 \
```
Comment (mmcgr) [inline src/Makefile.am:238]: Keep the backslash alignment in the Makefile. (Or set out in a bold new direction using `text[space][space]\` and remove all alignment.)

File: src/Makefile.am [hunk 2]
```diff
@@ -235,23 +235,23 @@ souffle_sources = \
...
+        ram/Node.h                                      \
+        ram/Operation.h                                 \
+        ram/Program.h                                   \
+        ram/Relation.h                                  \
+        ram/Statement.h                                 \
+        ram/TranslationUnit.h                           \
+        ram/Utils.h                                     \
+        ram/Visitor.h                                   \
+        ram/analysis/Analysis.h                         \
+        ram/analysis/ComplexityAnalysis.cpp             \
```
Comment (mmcgr) [inline src/Makefile.am:249]: `Analysis` could be removed from the filenames of the analyses in `ram/analysis`

Comment (b-scholz) [reply src/Makefile.am:249]: Let's do this in another step.

File: src/Makefile.am (hunk 1 above)
Comment (b-scholz) [reply src/Makefile.am:238]: I will do this.

File: src/Makefile.am (hunk 2 above)
Comment (b-scholz) [reply src/Makefile.am]: There are many files with not quite suitable filenames.

**Comment by mmcgr** [comment]
Relates to #462

**Pull request merged**